### PR TITLE
Add Binomial-Normal BMA Meta-Analysis

### PR DESCRIPTION
**Pull request by FBartos**
also now:
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/2129
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1201

### CODE REVIEW

File: R/bayesianmetaanalysisbinomial.R [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+  if (options[["priorDistributionPlot"]])
+    .bibmaPriorsPlots(jaspResults, options)
+
+
+  ### Inference
+  # default summary
+  .robmaSummaryTable(jaspResults, options, type = "BiBMA")
+  # models overview
+  if (options[["inferenceModelsOverview"]])
+    .robmaModelsOvervievTable(jaspResults, options, type = "BiBMA")
```
Comment (juliuspfadt) [inline R/bayesianmetaanalysisbinomial.R:50]:
```suggestion
    .robmaModelsOverviewTable(jaspResults, options, type = "BiBMA")
```

File: R/bayesianmetaanalysisbinomial.R [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+  # extract the priors
+  if (is.null(jaspResults[["model"]]))
+    priors <- jaspResults[["priors"]][["object"]]
+  else
+    priors <- jaspResults[["model"]][["object"]][["priors"]]
+
+  # create container for each of the parameters
+  for (parameter in c("effect", "heterogeneity")) {
+
+    if (!is.null(priorPlots[[parameter]]))
```
Comment (juliuspfadt) [inline R/bayesianmetaanalysisbinomial.R:220]: I would prefer that if you work with if else and one of them needs brackets that the other also gets brackets.

**Comment by juliuspfadt** [comment]
I am not at all done, some preliminary things maybe: 

- I get an empty plot with this file. I have no idea if what I was doing made sense, the only data set I could find was the metaAnalsyis stuff form the data library
[emptyplot.jasp.zip](https://github.com/jasp-stats/jaspMetaAnalysis/files/14195847/emptyplot.jasp.zip)
- I do not like the name of the analysis in JASP. Why not call it "Binomial Meta-Analysis"? The "Binomial" in brackets just makes me think what should be in brackets for the Meta Analysis above it. But maybe that is the actual issue. If one if Binomial what is the other?

File: inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+			}
+
+			CheckBox
+			{
+				label: 				qsTr("Effective sample size")
+				name:				"advancedAutofitEss"
+				checked:			true
+				childrenOnSameRow:	true
+
+				DoubleField
```
Comment (juliuspfadt) [inline inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:137]: I wonder if it makes sense to have this depend on the number of samples?

File: inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+				{
+					name:			"advancedAutofitEssTarget"
+					defaultValue:	500
+					min:			1
+					inclusive:		JASP.None
+				}
```
Comment (juliuspfadt) [inline inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:143]: I wonder if it makes sense to have this depend on the number of samples?

File: inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+						DropDown
+						{
+							name:			"advancedAutofitMaximumFittingTimeTargetUnit"
+							values:
+							[
+								{ label: qsTr("hours"),				value: "hours"},
+								{ label: qsTr("minutes"),			value: "mins"},
+								{ label: qsTr("seconds"),			value: "secs"}
+							]
+						}
```
Comment (juliuspfadt) [inline inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:205]: I am curious, how do you measure the time?

File: inst/qml/qml_components/RobustBayesianMetaAnalysisDiagnostics.qml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+			name:		"mcmcDiagnosticsPlotPeese"
+			visible:	analysisType === "RoBMA"
+		}
+	}
+
+	Group
+	{
+		Group
+		{
+			title:			qsTr("Type")
```
Comment (juliuspfadt) [inline inst/qml/qml_components/RobustBayesianMetaAnalysisDiagnostics.qml:80]: So are all these plots? Is it the plot type of the Effect and Heterogeneity plots? If that is the case it should be better visible.

**Review by juliuspfadt — CHANGES_REQUESTED** [review]
Very nice! 
See the comments for potential improvements

File: R/bayesianmetaanalysisbinomial.R (hunk 1 above)
Comment (FBartos) [reply R/bayesianmetaanalysisbinomial.R:50]: lol thanks!

File: R/bayesianmetaanalysisbinomial.R [hunk 7]
```diff
@@ -0,0 +1,457 @@
...
+  } else {
+    # only tries to update labels if they weren't supplied originally
+    fit <- update(fit, study_names = if (options[["studyLabel"]] != "") dataset[, options[["studyLabel"]]])
+
+  }
+
+
+  # error handling
+  if (jaspBase::isTryError(fit))
+    .quitAnalysis(fit)
```
[the remark this answers is not in the document]
Comment (FBartos) [reply R/bayesianmetaanalysisbinomial.R:449]: not really--this should never crash, but if it does, it at least gives users some package generated message to return

File: inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml (hunk 4 above)
Comment (FBartos) [reply inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:143]: yeah, that would make sense, but I believe we don't do it at other places and I wouldn't like to have the autofit options change automatically once I adjust the rest of MCMC settings

File: inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml (hunk 5 above)
Comment (FBartos) [reply inst/qml/qml_components/RobustBayesianMetaAnalysisAdvanced.qml:205]: `runjags` R package does that

File: inst/qml/qml_components/RobustBayesianMetaAnalysisDiagnostics.qml (hunk 6 above)
Comment (FBartos) [reply inst/qml/qml_components/RobustBayesianMetaAnalysisDiagnostics.qml:80]: I'm not sure what you mean actually

**Comment by FBartos** [comment]
I fixed everything but one unit test--it works well when running interactively but running it via testAnalysis always creates slightly different results. No idea what's the issue though (I updated all packages etc).

**Comment by FBartos** [comment]
the convergence check function does not refit the model, but the `update()` function in RoBMA does not update the convergence assessment properly (https://github.com/FBartos/RoBMA/issues/34)

I would just merge the PR and I will fix the package side bug separately.

**Review by juliuspfadt — APPROVED** [review]
sorry for taking so long

**Pull request merged**